src/util/thread.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "util_error.h"
+
+#include <windef.h>
+#include <winbase.h>
+
+namespace dxvk {
+  class thread {
+  private:
+    HANDLE m_handle;
+
+    template<typename T>
+    class thread_proc {
+    private:
+      T proc;
+    public:
+      thread_proc(T &proc) : proc(proc) {}
+
+      static DWORD WINAPI nativeProc(void *arg) {
+        thread_proc *proc = reinterpret_cast<thread_proc*>(arg);
+        proc->proc();
+        delete proc;
+        return 0;
+      }
+    };
+
+  public:
+    template<class T>
+    explicit thread(T &&func) {
+        thread_proc<T> *proc = new thread_proc<T>(func);
+        m_handle = CreateThread(NULL, 0, thread_proc<T>::nativeProc, proc, 0, NULL);
```

Comment:
Please use `nullptr` instead of `NULL` for consistency.

Reply:
Sure. I will also use ::CreateThread while I'm at this.